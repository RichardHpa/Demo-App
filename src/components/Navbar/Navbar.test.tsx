import { setMatchMediaMock } from 'tests/matchMediaMock'
import { render, screen } from 'tests/renderHelpers'

import { Navbar } from './Navbar'

describe('Navbar', () => {
  beforeEach(() => {
    setMatchMediaMock()
  })

  test('renders Navbar', () => {
    render(<Navbar />)

    const link = screen.getByRole('link', { name: /site logo/i })

    expect(link).toBeInTheDocument()

    expect(link).toMatchSnapshot()
  })

  test('renders all links', () => {
    render(<Navbar />)

    const homeLink = screen.getByRole('link', { name: /home/i })
    const aboutLink = screen.getByRole('link', { name: /about/i })

    expect(homeLink).toBeInTheDocument()
    expect(aboutLink).toBeInTheDocument()
  })
})
