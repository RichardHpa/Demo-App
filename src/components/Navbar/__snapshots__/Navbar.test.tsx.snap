// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar 1`] = `
<a
  class="mr-8 flex p-0 text-2xl tracking-tighter no-underline text-gray-700 dark:text-white"
  href="/"
>
  <span
    class="inline-block font-medium"
  >
    Site Logo
  </span>
</a>
`;
